### PR TITLE
refactor indexer 

### PR DESCRIPTION
**Pull request by MJedr**

### CODE REVIEW

File: backend/inspirehep/indexer/base.py [hunk 1]
```diff
@@ -72,6 +75,10 @@ def _process_bulk_record_for_index(
             "_version_type": version_type,
             "_source": self._prepare_record(record, index, doc_type),
         }
+        if current_app.config["FEATURE_FLAG_ENABLE_FULLTEXT"]:
+            ingestion_pipeline_name = current_app.config["ES_FULLTEXT_PIPELINE_NAME"]
+            payload["pipeline"] = ingestion_pipeline_name
```
Comment (drjova) [inline backend/inspirehep/indexer/base.py:80]: shouldn't this be only for literature?

File: backend/inspirehep/indexer/api.py [hunk 2]
```diff
@@ -21,6 +21,7 @@ def get_references_to_update(record):
             uuids_to_reindex |= (
                 record.get_all_connected_records_uuids_of_modified_collaborations()
             )
+        uuids_to_reindex |= record.get_modified_journal_uuids()
```
Comment (drjova) [inline backend/inspirehep/indexer/api.py:24]: shouldn't we have it for experiments, institutions etc?

Comment (drjova) [reply backend/inspirehep/indexer/api.py:24]: actually we just need to update this one instead https://github.com/inspirehep/inspirehep/blob/master/backend/inspirehep/records/api/literature.py#L555-L575

Comment (MJedr) [reply backend/inspirehep/indexer/api.py:24]: probably we should

Comment (drjova) [reply backend/inspirehep/indexer/api.py]: As we discussed let's move this into `record.get_linked_papers_if_reference_changed`

**Review by drjova — APPROVED** [review]

**Pull request merged**